locator/token_metadata.cc

```diff
@@ -1261,13 +1259,30 @@ topology::topology(config cfg)
     _pending_locations[utils::fb_utilities::get_broadcast_address()] = std::move(cfg.local_dc_rack);
 }
 
-topology::topology(const topology& other)
-        : _dc_endpoints(other._dc_endpoints)
-        , _dc_racks(other._dc_racks)
-        , _current_locations(other._current_locations)
-        , _pending_locations(other._pending_locations)
-        , _sort_by_proximity(other._sort_by_proximity)
-{
+future<> topology::copy_gently(const topology& other) {
+    co_await clear_gently();
```

Comment:
This destructive form of copying is error-prone. copy_gently() doesn't say where we're copying to.

I prefer the future<topology> clone_gently() form where it's clear where things are going. It can call a private constructor from all the components (e.g. at the end `co_return topology(std::move(dc_endpoints), std::move(dc_racks), ... sort_by_proximity);`

Reply:
Sure, no problem.